test/integration/controller/core/resourceflavor_controller_test.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package core
+
+import (
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha1"
+	"sigs.k8s.io/kueue/pkg/controller/core"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	"sigs.k8s.io/kueue/test/integration/framework"
+)
+
+// +kubebuilder:docs-gen:collapse=Imports
+
+var _ = ginkgo.Describe("ResourceFlavor controller", func() {
+	var ns *corev1.Namespace
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-resourceflavor-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(framework.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.Context("When no clusterQueue references resourceFlavors", func() {
+		var resourceFlavor *kueue.ResourceFlavor
+		ginkgo.BeforeEach(func() {
+			resourceFlavor = utiltesting.MakeResourceFlavor("cq-refer-resourceflavor").Obj()
+			gomega.Expect(k8sClient.Create(ctx, resourceFlavor)).To(gomega.Succeed())
+			// Wait for the resourceFlavor to add finalizer.
+			time.Sleep(300 * time.Millisecond)
+		})
+
+		ginkgo.AfterEach(func() {
+			gomega.Expect(framework.DeleteResourceFlavor(ctx, k8sClient, resourceFlavor)).To(gomega.Succeed())
+		})
+
+		ginkgo.It("Should have finalizer in new created resourceFlavor", func() {
+			var rf kueue.ResourceFlavor
+			gomega.Eventually(func() []string {
+				gomega.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(resourceFlavor), &rf)).To(gomega.Succeed())
+				return rf.GetFinalizers()
+			}, framework.Timeout, framework.Interval).Should(utiltesting.Equal([]string{core.ResourceFlavorFinalizerName}))
+		})
+
+		ginkgo.It("Should update resourceFlavor successfully", func() {
```

Comment:
the controller doesn't do updates. Maybe it should say something like:

Should handle resourceFlavor updates